mssql_rawquery_block.py

```diff
@@ -0,0 +1,50 @@
+from nio.properties import Property, VersionProperty, StringProperty
+from nio.block.terminals import output
+from .mssql_base import MSSQLBase
+from nio.block.mixins.enrich.enrich_signals import EnrichSignals
+
+@output('results', label='Results')
+@output('no_results', label='No Results')
+class MSSQLRawQuery(EnrichSignals, MSSQLBase):
+
+    version = VersionProperty("1.0.0")
+    query = StringProperty(title='Parameterized Query (use ? for any user-supplied values)', default='SELECT * FROM table where id=?', order=10)
+    parameters = Property(title='Substitution Parameters (As List, In Order)', default='{{[]}}', order=11)
+
+    def process_signals(self, signals):
+        if self.isConnecting:
+            self.logger.error('Connection already in progress. Dropping signals.')
+        else:
+            try:
+                cursor = self.cnxn.cursor()
+            except Exception as e:
+                self.disconnect()
+                self.connect()
+                cursor = self.cnxn.cursor()
+
+            output_signals = []
+
+            for signal in signals:
+                _query = self.query(signal)
+                _parameters = self.parameters(signal)
+
+                result = cursor.execute(_query, _parameters) if len(_parameters) > 0 else cursor.execute(_query)
+
+                try:
+                    rows = result.fetchall()

```

Comment:
this is more of an interesting/philosophical question: given the result-set is _huge_ is doing a `fetchall()` memory safe. or should it self partition into multiple [`fetchmany()`](https://www.python.org/dev/peps/pep-0249/#fetchmany) and dispatch them in chunks.

Reply:
Assuming they can use the `LIMIT` operator in the SQL statement I'm ok with `fetchall()` for now. It kind of leaves the pagination/cursoring up to the block configurer. I wouldn't be opposed to a "paged" block though that took a cursor/page size and did this for the user.

Reply:
In the service implementation, I've set an env var to limit GETs to 1000 unless otherwise specified